### ISSUE TITLE
[BUG] PHP7.3 installation issues.

### ISSUE THREAD

**Issue by boxxroom** [sprintcube/docker-compose-lamp#244]
**Describe the bug**
New `git clone ` installation. Rename and update the `.env` file to use php73, mariadb106

**which Branch / PHP Version are you using ?**
main

**Steps to reproduce**
1. Git Clone
2. rename sample.env to .env
3. Use php73 as the PHP version, update to use mariadb106 as database
4. docker compose up -d

**Expected behaviour**
Successful installation of docker with PHP 73

**Screenshots**

**Desktop (please complete the following information):**
 - OS: MacOS (Intel)
 - Docker Version: 4.19.0
```
docker -v
Docker version 23.0.5, build bc4487a
```
**Additional context**
```shell
 => ERROR [ 2/11] RUN apt-get -y update --fix-missing &&     apt-get upgrade -y &&     apt-get --no-install-recommends install -y apt-utils &&     r  8.7s
------
 > [ 2/11] RUN apt-get -y update --fix-missing &&     apt-get upgrade -y &&     apt-get --no-install-recommends install -y apt-utils &&     rm -rf /var/lib/apt/lists/*:
#0 0.781 Ign:1 http://deb.debian.org/debian stretch InRelease
#0 0.781 Ign:2 http://security.debian.org/debian-security stretch/updates InRelease
#0 0.799 Ign:3 http://deb.debian.org/debian stretch-updates InRelease
#0 0.807 Get:4 http://security.debian.org/debian-security buster/updates InRelease [34.8 kB]
#0 0.853 Get:5 http://deb.debian.org/debian buster InRelease [122 kB]
#0 0.933 Ign:6 http://security.debian.org/debian-security stretch/updates Release
#0 0.977 Ign:7 http://security.debian.org/debian-security stretch/updates/main amd64 Packages
#0 0.987 Get:8 http://deb.debian.org/debian buster-updates InRelease [56.6 kB]
#0 1.038 Ign:9 http://security.debian.org/debian-security stretch/updates/main all Packages
#0 1.057 Ign:10 http://deb.debian.org/debian stretch Release
#0 1.075 Ign:11 http://deb.debian.org/debian stretch-updates Release
#0 1.077 Ign:7 http://security.debian.org/debian-security stretch/updates/main amd64 Packages
#0 1.234 Ign:12 http://deb.debian.org/debian stretch/main all Packages
#0 1.235 Ign:9 http://security.debian.org/debian-security stretch/updates/main all Packages
#0 1.277 Ign:7 http://security.debian.org/debian-security stretch/updates/main amd64 Packages
#0 1.295 Get:13 http://security.debian.org/debian-security buster/updates/main amd64 Packages [521 kB]
#0 1.317 Ign:14 http://deb.debian.org/debian stretch/main amd64 Packages
#0 1.568 Ign:15 http://deb.debian.org/debian stretch-updates/main amd64 Packages
#0 1.759 Ign:16 http://deb.debian.org/debian stretch-updates/main all Packages
#0 1.773 Ign:12 http://deb.debian.org/debian stretch/main all Packages
#0 1.782 Get:17 http://deb.debian.org/debian buster/main amd64 Packages [7909 kB]
#0 1.782 Ign:9 http://security.debian.org/debian-security stretch/updates/main all Packages
#0 1.937 Ign:7 http://security.debian.org/debian-security stretch/updates/main amd64 Packages
#0 2.101 Ign:9 http://security.debian.org/debian-security stretch/updates/main all Packages
#0 2.316 Ign:7 http://security.debian.org/debian-security stretch/updates/main amd64 Packages
#0 2.421 Ign:9 http://security.debian.org/debian-security stretch/updates/main all Packages
#0 2.657 Err:7 http://security.debian.org/debian-security stretch/updates/main amd64 Packages
#0 2.657   404  Not Found [IP: 151.101.194.132 80]
#0 2.839 Ign:9 http://security.debian.org/debian-security stretch/updates/main all Packages
#0 7.037 Ign:14 http://deb.debian.org/debian stretch/main amd64 Packages
#0 7.037 Get:18 http://deb.debian.org/debian buster-updates/main amd64 Packages [8788 B]
#0 7.095 Ign:15 http://deb.debian.org/debian stretch-updates/main amd64 Packages
#0 7.113 Ign:16 http://deb.debian.org/debian stretch-updates/main all Packages
#0 7.134 Ign:12 http://deb.debian.org/debian stretch/main all Packages
#0 7.178 Ign:14 http://deb.debian.org/debian stretch/main amd64 Packages
#0 7.226 Ign:15 http://deb.debian.org/debian stretch-updates/main amd64 Packages
#0 7.260 Ign:16 http://deb.debian.org/debian stretch-updates/main all Packages
#0 7.277 Ign:12 http://deb.debian.org/debian stretch/main all Packages
#0 7.319 Ign:14 http://deb.debian.org/debian stretch/main amd64 Packages
#0 7.365 Ign:15 http://deb.debian.org/debian stretch-updates/main amd64 Packages
#0 7.525 Ign:16 http://deb.debian.org/debian stretch-updates/main all Packages
#0 7.558 Ign:12 http://deb.debian.org/debian stretch/main all Packages
#0 7.599 Ign:14 http://deb.debian.org/debian stretch/main amd64 Packages
#0 7.650 Ign:15 http://deb.debian.org/debian stretch-updates/main amd64 Packages
#0 7.671 Ign:16 http://deb.debian.org/debian stretch-updates/main all Packages
#0 7.692 Ign:12 http://deb.debian.org/debian stretch/main all Packages
#0 7.741 Err:14 http://deb.debian.org/debian stretch/main amd64 Packages
#0 7.741   404  Not Found
#0 7.788 Err:15 http://deb.debian.org/debian stretch-updates/main amd64 Packages
#0 7.788   404  Not Found
#0 7.806 Ign:16 http://deb.debian.org/debian stretch-updates/main all Packages
#0 7.916 Fetched 8652 kB in 7s (1191 kB/s)
#0 7.916 Reading package lists...
#0 8.662 W: The repository 'http://security.debian.org/debian-security stretch/updates Release' does not have a Release file.
#0 8.662 W: The repository 'http://deb.debian.org/debian stretch Release' does not have a Release file.
#0 8.662 W: The repository 'http://deb.debian.org/debian stretch-updates Release' does not have a Release file.
#0 8.662 E: Failed to fetch http://security.debian.org/debian-security/dists/stretch/updates/main/binary-amd64/Packages  404  Not Found [IP: 151.101.194.132 80]
#0 8.662 E: Failed to fetch http://deb.debian.org/debian/dists/stretch/main/binary-amd64/Packages  404  Not Found
#0 8.662 E: Failed to fetch http://deb.debian.org/debian/dists/stretch-updates/main/binary-amd64/Packages  404  Not Found
#0 8.662 E: Some index files failed to download. They have been ignored, or old ones used instead.
------
failed to solve: process "/bin/sh -c apt-get -y update --fix-missing &&     apt-get upgrade -y &&     apt-get --no-install-recommends install -y apt-utils &&     rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
```

**Comment by bramus** [comment]
That’s because Debian `stretch` has been abandoned. Easiest fix is to upgrade to `buster`. Don’t forget to update the _“Install useful tools and install important libaries”_ section too – see the `php74/Dockerfile` for inspiration.

Your resulting `php73/Dockerfile` should start like this:

```
FROM php:7.3-apache-buster

# Surpresses debconf complaints of trying to install apt packages interactively
# https://github.com/moby/moby/issues/4032#issuecomment-192327844
 
ARG DEBIAN_FRONTEND=noninteractive

# Update
RUN apt-get -y update --fix-missing && \
    apt-get upgrade -y && \
    apt-get --no-install-recommends install -y apt-utils && \
    rm -rf /var/lib/apt/lists/*


# Install useful tools and install important libaries
RUN apt-get -y update && \
    apt-get -y --no-install-recommends install nano wget \
dialog \
libsqlite3-dev \
libsqlite3-0 && \
    apt-get -y --no-install-recommends install default-mysql-client \
zlib1g-dev \
libzip-dev \
libicu-dev && \
    apt-get -y --no-install-recommends install --fix-missing apt-utils \
build-essential \
git \
curl \
libonig-dev && \ 
    apt-get install -y iputils-ping && \
    apt-get -y --no-install-recommends install --fix-missing libcurl4 \
libcurl4-openssl-dev \
zip \
openssl && \
    rm -rf /var/lib/apt/lists/* && \
    curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer

# Install xdebug
…
```

**Comment by sebsebseb123** [comment]
Thanks for the answer @bramus 
...where was this 6 hours ago!? Haha. I just spent 6 hrs trying to debug this. But at least I learned a lot about docker and sources.list files and such.

I solved it by updating the sources.list file to the archived sources. I didn't want to change the OS version in case any of the subsequent setup/config relied on "stretch".

So, in the php73 Dockerfile in `./bin/php73` I added these lines after `FROM php:7.3-apache-stretch` :

```
# Remove the old sources.list file.
RUN rm /etc/apt/sources.list

# Replace with this list.
RUN echo 'deb http://archive.debian.org/debian/ stretch main contrib non-free\ndeb http://archive.debian.org/debian/ stretch-proposed-updates main contrib non-free\ndeb http://archive.debian.org/debian-security stretch/updates main contrib non-free' >> /etc/apt/sources.list
```

I found these sources from https://www.geekersdigest.com/apt-etc-apt-sources-list-for-every-debian-release/

docker compose works for me now 😊

**Comment by boxxroom** [comment]
Thanks both for the answers here @bramus  @sebsebseb123 - I will try both of them and let you know the results.

Before posting this issue, I did start to merge some PHP74/Dockerfile conf in to PHP73/Dockerfile after reading `stretch` had been abandoned but came across installation errors when running `docker compose up -d`.

I believe this issue also occurs on PHP72 and possible earlier versions. Would it be a case of using fix but with `FROM php:7.2-apache-buster`

@bramus @sebsebseb123 with this issue being a core issue within the repo, it might worth creating a pull-request for this fix.

**Comment by MrOffline77** [comment]
Hi Guys, 
thanks for bringing this up. And sorry for the delay. I just added a fix with #248. Please reopen the issue if there is still an issue.
